pkg/webui/console/views/gateway-general-settings/index.js

```diff
@@ -90,131 +81,100 @@ export default class GatewayGeneralSettings extends React.Component {
-
-    let changed
-    if (Object.keys(idsDiff).length) {
-      changed = { ids: idsDiff, ...entityDiff }
-    } else {
-      changed = entityDiff
-    }
-
-    const update =
-      'attributes' in changed
-        ? { ...changed, attributes: mapFormValueToAttributes(values.attributes) }
-        : changed
-
-    try {
-      await updateGateway(gtwId, update)
-      this.formRef.current.resetForm()
-      toast({
-        title: gtwId,
-        message: m.updateSuccess,
-        type: toast.types.SUCCESS,
-      })
-    } catch (error) {
-      this.formRef.current.resetForm({ values })
-      await this.setState({ error })
-    }
+  async handleSubmitSuccess() {
+    const { gateway } = this.props
+
+    const {
+      ids: { gateway_id: gatewayId },
+    } = gateway
+
+    toast({
+      title: gatewayId,
+      message: m.updateSuccess,
+      type: toast.types.SUCCESS,
+    })
   }
 
   @bind
   async handleDelete() {
-    const { gtwId, deleteGateway, onDeleteSuccess } = this.props
+    const { gtwId, deleteGateway } = this.props
 
-    await this.setState({ error: '' })
+    return deleteGateway(gtwId)
+  }
+
+  @bind
+  async handleDeleteSuccess() {
+    const { gateway, onDeleteSuccess } = this.props
+    const {
+      ids: { gateway_id: gatewayId },
+    } = gateway
 
-    try {
-      await deleteGateway(gtwId)
-      onDeleteSuccess()
-    } catch (error) {
-      this.formRef.current.setSubmitting(false)
-      this.setState({ error })
-    }
+    onDeleteSuccess()
+    toast({
+      title: gatewayId,
+      message: m.deleteSuccess,
+      type: toast.types.SUCCESS,
+    })
   }
 
-  render() {
-    const { gateway, gtwId } = this.props
-    const { error } = this.state
+  @bind
+  async handleDeleteFailure() {
+    const { gateway } = this.props
     const {
-      ids,
-      gateway_server_address,
-      frequency_plan_id,
-      enforce_duty_cycle,
-      name,
-      description,
-      location_public,
-      status_public,
-      schedule_downlink_late,
-      update_location_from_status,
-      auto_update,
-      update_channel,
-      schedule_anytime_delay,
-      attributes,
+      ids: { gateway_id: gatewayId },
     } = gateway
 
-    const initialValues = {
-      ids: { ...ids },
-      gateway_server_address,
-      frequency_plan_id,
-      enforce_duty_cycle,
-      name,
-      description,
-      location_public,
-      status_public,
-      schedule_downlink_late,
-      update_location_from_status,
-      auto_update,
-      update_channel,
-      schedule_anytime_delay,
-      attributes: mapAttributesToFormValue(attributes),
-    }
+    toast({
+      title: gatewayId,
+      message: m.deleteFailure,
+      type: toast.types.ERROR,
+    })
+  }
+
+  render() {
+    const { gateway } = this.props
 
     return (
       <Container>
-        <PageTitle title={sharedMessages.generalSettings} />
+        <PageTitle className={style.visuallyHidden} title={sharedMessages.generalSettings} />
```

Comment:
Theres a `hideHeading` prop to do just that.

```suggestion
        <PageTitle title={sharedMessages.generalSettings} hideHeading />
```